docs/recipes/unfold-promises.md

```diff
@@ -0,0 +1,37 @@
+# Unfold Promises
+
+In this example we use `most.unfold` to help deal with promises.
+
+We build up a stream of urls by providing `most.unfold` creator with a seed
+or an array of urls.
+
+The unfolding will continue building up the stream each time removing one value
+off the array, invoking the unfold function with the remaining values of the array.
+
+Once the array is empty `tuple.done` becomes true and the unfolding will stop.
+
+The interesting part is that we [fetch](https://fetch.spec.whatwg.org) one url
+at a time, process it before returning a `tuple` with a new seed.
+
+If we were to run the code below we would observe the 1st url gets processed
+instantly while the 2nd waits for about 3 seconds before continuing to resolve
+the last.
+
+```js
+var most = require('most');
+var urls = ['http://reqr.es/api/users?page=2', 
+            'http://reqr.es/api/users?delay=3', 
+            'http://reqr.es/api/users?page=3'];
+
+most.unfold(function (urls) {
+    return urls.length === 0 ? {
+        done: true
+    } : fetch(urls[0]).then(function(content){
```

Comment:
This is a tricky thing to format.  Took me a minute to parse it.  Maybe something like:

```js
return urls.length === 0
    ? { done: true }
    : fetch(urls[0]).then(function(content) {
```

I dunno, what do you think?